sdk/src/azure/core/az_http_request.c

```diff
@@ -170,55 +119,16 @@ az_http_request_set_query_parameter(az_http_request* ref_request, az_span name,
   // name or value can't be empty
   _az_PRECONDITION(az_span_size(name) > 0 && az_span_size(value) > 0);
 
-  // check if query parameter is already in url
-  int32_t pre_existing_query_parameter_start_index = 0;
-  az_span pre_existing_query_parameter_value = _az_http_request_find_query_parameter(
-      ref_request, name, &pre_existing_query_parameter_start_index);
   az_span url_remainder
       = az_span_slice_to_end(ref_request->_internal.url, ref_request->_internal.url_length);
 
-  if (pre_existing_query_parameter_start_index > 0)
-  { // a negative difference means shifting left and no required length.
-    // a positive difference means new value will require shifting right.
-    int32_t pre_existing_query_parameter_value_size
-        = az_span_size(pre_existing_query_parameter_value);
-    int32_t difference = az_span_size(value) - pre_existing_query_parameter_value_size;
-    int32_t required_length = difference > 0 ? difference : 0;
-    AZ_RETURN_IF_NOT_ENOUGH_SIZE(url_remainder, required_length);
-
-    // There is already a query parameter name. update the value if different
-    if (az_span_is_content_equal(value, pre_existing_query_parameter_value))
-    {
-      // No need to do anything, existing value is equal to the new one
-      return AZ_OK;
-    }
-
-    // Replace the value content. This might shift right or left the url contents.
-    // No need to check
-    AZ_RETURN_IF_FAILED(_az_span_replace(
-        ref_request->_internal.url,
-        ref_request->_internal.url_length,
-        pre_existing_query_parameter_start_index,
-        pre_existing_query_parameter_start_index + pre_existing_query_parameter_value_size,
-        value));
-    ref_request->_internal.url_length += difference;
-    if (difference < 0)
-    {
-      // clear anything after a left shift
-      difference *= -1;
-      az_span_fill(
-          az_span_slice(
-              az_span_slice_to_end(ref_request->_internal.url, ref_request->_internal.url_length),
-              0,
-              difference),
-          0);
-    }
-    return AZ_OK;
-  }
-
-  // Adding new query parameter. Adding +2 to required length to include extra required symbols `=`
+  // Adding query parameter. Adding +2 to required length to include extra required symbols `=`
   // and `?` or `&`.
-  int32_t required_length = az_span_size(name) + az_span_size(value) + 2;
+  int32_t required_length
+      = (is_query_url_encoded
+             ? (az_span_size(name) + az_span_size(value))
```

Comment:
Where's the +2 for this? If this is a bug, we should add a test that catches this issue of incorrect `required_length` before fixing it.

Reply:
Oh, never mind. The `+2` below applies to both conditions.

Maybe change it to: `required_length = 2 + (...)`